src/App.tsx

```diff
@@ -1,24 +1,230 @@
+        setTodos(prevState => prevState
+          .map(currentTodo => (
+            currentTodo.id !== updatedTodo.id
+              ? currentTodo
+              : updatedTodo
+          )));
+      });
+  };
+
+  const handelClearComplited = () => {
+    const allCompleatedTodos = todos.filter(todo => todo.completed === true);
+
+    allCompleatedTodos.forEach(todo => (
+      setIsLoading(prevArray => [...prevArray, todo.id])
+    ));
+
+    Promise.all(allCompleatedTodos.map(todo => TodoSevise.deleteTodo(todo.id)))
+      .then(() => setTodos(todos.filter(todo => !todo.completed)))
+      .catch(() => (
+        setErrorMessage('Unable to delete a todo')
+      ))
+      .finally(() => {
+        setIsLoading([]);
+        setIsDisabled(false);
+      });
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          onToddoAdd={handleAddTodo}
+          onToggleChange={handleToggleChange}
+          todos={visibleTodos}
+          inputRef={inputRef}
+          setTempTodo={setTempTodo}
+          setIsDisabled={setIsDisabled}
+          setErrorMessage={setErrorMessage}
+          todoTitle={todoTitle}
+          setTodoTitle={setTodoTitle}
+          isDisabled={isDisabled}
+        />
+        <section className="todoapp__main" data-cy="TodoList">
+          {visibleTodos.map(todo => (
+            <TodoItem
+              todo={todo}
+              onTodoDelete={handleDeleteTodo}
+              isLoading={isLoading}
+              handleLoading={handleLoading}
+              key={todo.id}
+              onTodoUpdate={handleUpdateTodo}
+              onToggleChange={handleToggleChange}
+            />
+          ))}
+
+          {(tempTodo) && (
+            <TodoItem
+              todo={tempTodo}
+              onTodoDelete={handleDeleteTodo}
+              isLoading={isLoading}
+              handleLoading={handleLoading}
+              onTodoUpdate={handleUpdateTodo}
+              onToggleChange={handleToggleChange}
+              key={tempTodo.id}
+            />
+          )}
+        </section>
+
+        {Boolean(todos.length)
+          && (
+            <Footer
+              onFilterStatus={handleFilterStatus}
+              todosFilterStatus={status}
+              onDeleteCompleated={handelClearComplited}
+              hasCompleted={hasCompleted}
+              activeTodosCount={activeTodosCount}
+            />
+          )}
+      </div>
+
+      <div
+        data-cy="ErrorNotification"
+        className={classNames(
```

Comment:
Create an `ErrorNotification` component, and also move related logic to it too, this one:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/63243136/4e856d8e-cf15-4e88-8d42-112d025a6597)